classic-ui/view and layers

MrTango opened the pull request.

MrTango committed:
Start updating existing view chapter

MrTango committed:
Start updating existing view chapter

stevepiercy commented:
@MrTango before getting too far, please make sure that you bring this PR up to date with `6-dev`, either with a merge or rebase.

MrTango committed:
improve classic-ui/views chapter

MrTango committed:
improve classic-ui/views chapter

MrTango committed:
cleanup unsused menu items and sort classic-ui items

MrTango committed:
imrpove classic-ui/views and add layers

MrTango committed:
add graphviz support and use it in classic-ui/views

MrTango commented:
It`s far from perfect, but we have to start somewhere.
We should merge this after review and continue to improve it and also fill in some missing peaces to fix the references to other pages.

stevepiercy commented on docs/classic-ui/layers.md:
```diff
@@ -0,0 +1,292 @@
+instance functionality in your site at a time, while still allowing you
+to have possibly conflicting add-on products in your buildout and
+ZCML. Remember that multiple Plone site instances can share
+the same ZCML and code files.
```
The last sentence does not connect with the first. Why is it necessary to remember this? And where was the concept introduced that the reader should remember it?

stevepiercy commented on docs/classic-ui/layers.md:
```diff
@@ -0,0 +1,292 @@
+the same ZCML and code files.
+
+Many ZCML directives take the optional `layer` parameter. See example,
+[resourceDirectory](http://apidoc.zope.org/++apidoc++/ZCML/http_co__sl__sl_namespaces.zope.org_sl_browser/resourceDirectory/index.html)
```
This link is a 404. I don't know what is a valid URL for this.

stevepiercy commented on docs/classic-ui/layers.md:
```diff
@@ -0,0 +1,292 @@
+
+If you want to override a view or a viewlet unconditionally for all sites
+without the add-on product installer
+support you need to use `overrides.zcml`.
```
How do I use this?

stevepiercy replied:
See #1426

ksuess commented:
> fill in some missing peaces

I love your typo!

stevepiercy committed:
Merge 6-dev

stevepiercy committed:
Add graphviz configuration

stevepiercy committed:
Remove plone-restapi.md from backend, as it has been promoted to the top level navigation

stevepiercy committed:
Reorder pages to be alphabetic by page title, not file name.

stevepiercy committed:
Merge of 6-dev

stevepiercy committed:
Clean up layers.md, add interface to Glossary

stevepiercy committed:
Blackify

stevepiercy committed:
Clean up glossary.md

stevepiercy committed:
Add instructions to install Graphviz, move note below Enchant install.

stevepiercy committed:
Commit WIP

stevepiercy committed:
Add section heading labels.
One sentence per line.

stevepiercy commented on docs/classic-ui/views.md:
````diff
@@ -1,13 +1,1007 @@
+`configure.zcml` to use `browser` configuration directives.
+```
+
+The view in question is registered against a {ref}`classic-ui-layers-label`, it will be available after restart, after you run {doc}`Add/remove in Site setup </develop/addons/components/genericsetup>`.
````
This reference generates a Sphinx warning and must be fixed before merging. The target link to https://docs.plone.org/develop/addons/components/genericsetup does not make sense to me, otherwise I would replace it with that. Can you suggest a better target?

stevepiercy commented on docs/classic-ui/views.md:
````diff
@@ -1,13 +1,1007 @@
+Fix references in this section
+```
+
+Available {doc}`slot </adapt-and-extend/theming/templates_css/template_basics>`
````
This reference must be fixed before merge. The target goes to the page, not any specific section in the page. https://docs.plone.org/adapt-and-extend/theming/templates_css/template_basics

stevepiercy commented on docs/classic-ui/views.md:
```diff
@@ -1,13 +1,1007 @@
+
+`content`
+
+: render edit border yourself
```
What is this? Please elaborate.

stevepiercy commented on docs/classic-ui/views.md:
```diff
@@ -1,13 +1,1007 @@
+The Python constructor method of the view, `__init__()`, is special.
+You should almost never try to put your code there. Instead, use the `_call__()` method or further helper methods called from it.
+
+The `__init__()` method of the view might not have an {doc}`acquisition chain </develop/plone/serving/traversing>`
```
Fix this link before merging. I don't know what would be a good target, else I would do it myself.

stevepiercy committed:
Fix paragraphs in list items.

stevepiercy committed:
Add browser layer to Glossary as synonym to Layer

stevepiercy committed:
Temporary fix to silence warnings.
See permanent fix in https://github.com/plone/documentation/issues/1312

stevepiercy committed:
Provide temporary links to 5.2 docs instead of dead references.

stevepiercy commented on docs/classic-ui/views.md:
````diff
@@ -1,13 +1,1007 @@
+
+```{Note}
+Using `RestrictedPython` scripts (creating Python through the
+Management Interface) and Zope 2 Extension modules is discouraged.
````
"Zope 2"? Aren't we using Zope 5 now?

stevepiercy committed:
This time, really fix paragraphs in list items.

stevepiercy commented on docs/classic-ui/views.md:
````diff
@@ -1,13 +1,1007 @@
+  times.
+
+```{Note}
+The Plone 2.x way of doing this with TAL template language macros is
````
"Plone 2.x"? Is this note even relevant for Plone 6? Can this note be deleted?

MrTango replied:
Since we marcos are still part of the stack and being used in core, the statement should stay.
But it needs rephrasing.

stevepiercy commented on docs/classic-ui/views.md:
```diff
@@ -1,13 +1,1007 @@
+<div tal:replace="structure view/renderSummary" />
+
+<div class="description">
+    <div tal:condition="python:context.Description().decode('utf-8') != u'None'" tal:replace="structure context/Description" />
```
Are we still using markers for Unicode strings in Python 3? Can this snippet be updated for Python 3?

stevepiercy commented on docs/classic-ui/views.md:
```diff
@@ -1,13 +1,1007 @@
+- Zope  [BrowserView source code](https://github.com/zopefoundation/zope.publisher/blob/master/src/zope/publisher/browser.py).
+- [Five version](https://github.com/zopefoundation/Zope/blob/master/src/Products/Five/browser/__init__.py).
+  [Products.Five](https://github.com/zopefoundation/Zope/blob/master/src/Products/Five/doc/manual.txt) is a way to access some Zope 3 technologies from the Zope
+  2 codebase, which is used by Plone.
```
Is this true in Plone 6? Zope 2 and 3?

MrTango replied:
In general the info still makes sense. Zope 3 is completely different to Zope2 and now Zope4.x and yes i know :laughing:. 
The five package integrates Zope 3 technologies into Zope2/Zope4 or let's just say Zope from now on.
In other places where i found Zope 2 or Zope 3 i removed the 2 or 3, but it depends on the context.
Here i would keep it.

stevepiercy commented on docs/classic-ui/views.md:
```diff
@@ -1,13 +1,1007 @@
+- The `provider:` TAL expression.
+- Other Plone-specific TAL expression functions like `test()`.
+- Usually, Plone code needs the Five version of `ViewPageTemplateFile`.
+- Some subsystems, notably the `z3c.form` package, expect the Zope 3
```
Zope 3 again. Still needed in Plone 6?

stevepiercy committed:
Clean up English and MyST syntax and grammar.
Replace some non-functional links with links to the 5.2 docs. No broken links allowed!
Add Zope Component Architecture to Glossary term as alias for ZCA.

stevepiercy reviewed (changes requested):
Whew! I pushed up a lot of changes, mostly clean up of English and MyST syntax and grammar, but there are still some questions.

I also am absolutely not in favor of merging anything with a broken link. I put in temporary fixes and added a TODO for most, but there remain a few that I have no clue as to what the original intended target should be.

Finally I asked some questions in my comments about some content that appears to be ancient (Plone 2 and Zope 2, RLY?). Was this content copy-pasted without careful review and testing? We should always review anything that gets into a PR before merging it into the docs. No cruft allowed!

stevepiercy committed:
Update submodules/plone.api submodules/plone.restapi submodules/volto tips

stevepiercy committed:
Resolve conflicts with 6-dev in glossary.md

stevepiercy committed:
Resolve conflicts with 6-dev in conf.py

stevepiercy committed:
Merge branch '6-dev' into MrTango-views-templates

stevepiercy commented:
@MrTango I would appreciate a review and responses to my questions. Thank you!

stevepiercy commented:
Netlify was temporarily unable to reach GitHub. See preview links:

- https://deploy-preview-1291--6-dev-docs-plone-org.netlify.app/classic-ui/views.html
- https://deploy-preview-1291--6-dev-docs-plone-org.netlify.app/classic-ui/layers.html

MrTango committed:
fix broken link to layers example

MrTango committed:
Fix broken link to GenericSetup, add a placeholder for config-registry docs

MrTango committed:
Use Zope instead of Zope 2

MrTango committed:
cleanup some old namings

MrTango committed:
Merge branch '6-dev' into MrTango-views-templates

MrTango committed:
improve template slots section

MrTango committed:
Merge branch 'MrTango-views-templates' of https://github.com/plone/documentation into MrTango-views-templates

MrTango committed:
Fix broken links, add placeholder traversing chapter,

MrTango commented:
@stevepiercy i hope it is much better now, we should merge this soon and improve things later on. I'll spend some time during the sprint in Innsbruck in Feb on docs. There is still a lot to do.

jensens commented:
Can we get rid of all "prerendered for Plone version 4.x or greater."?

stevepiercy commented:
Please give me a few days to review.

stevepiercy committed:
Rename to Configuration Registry. Always spell out words.
Add meta information.

stevepiercy committed:
docs.plone.org -> 5.docs.plone.org

stevepiercy committed:
Improve meta information

stevepiercy committed:
Add spelling words

stevepiercy committed:
Improve meta information.
Spell out ZCML in its first appearance.
Add links to issue for todo.
Add introductory text to naked section headings.

stevepiercy committed:
Improve meta information.
Rename to align with title.

stevepiercy committed:
Rename and resort

stevepiercy committed:
Tidy up the theming section

stevepiercy committed:
Sort

stevepiercy committed:
Standardize formatting and styles
Put installation options for each operating system into tabs

stevepiercy committed:
Add meta information, tidy

stevepiercy committed:
Add view glossary entry

stevepiercy committed:
Linkcheck

stevepiercy committed:
Fix syntax and meta information

stevepiercy reviewed (dismissed):
I completed my review and pushed commits.

I fixed a few broken links and syntax errors elsewhere while I was at it.

When CI is green, I will merge this PR. Thank you!

stevepiercy committed:
Fix spelling

stevepiercy committed:
Fix spelling

stevepiercy committed:
per @jensens

stevepiercy commented:
All green. Thank you, @MrTango! What a huge effort!
